From 1152f7eaecdebe8043224e53106e7b15644796cf Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Wed, 16 Mar 2022 14:43:43 +0530
Subject: [PATCH] Fix build error seen with clang14

Use of clang-14 gives below build error:
error: bitwise or with non-zero value always evaluates
to true [-Werror,-Wtautological-bitwise-compare]

Signed-off-by: svenate <salini.venate@intel.com>
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 omx_components/src/mfx_omx_venc_component.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/omx_components/src/mfx_omx_venc_component.cpp b/omx_components/src/mfx_omx_venc_component.cpp
index 5b81eab..f13d544 100755
--- a/omx_components/src/mfx_omx_venc_component.cpp
+++ b/omx_components/src/mfx_omx_venc_component.cpp
@@ -1679,7 +1679,11 @@ OMX_ERRORTYPE MfxOmxVencComponent::SetParameter(
                     {
                         MFX_OMX_AUTO_TRACE_I32(pParam->bStoreMetaData);
                         {
-                            m_MfxVideoParams.IOPattern &= !(MFX_IOPATTERN_IN_VIDEO_MEMORY | MFX_IOPATTERN_IN_SYSTEM_MEMORY | MFX_IOPATTERN_IN_OPAQUE_MEMORY);
+                            if (MFX_IOPATTERN_IN_VIDEO_MEMORY || MFX_IOPATTERN_IN_SYSTEM_MEMORY || MFX_IOPATTERN_IN_OPAQUE_MEMORY)
+                                m_MfxVideoParams.IOPattern = 0;
+                            else
+                               m_MfxVideoParams.IOPattern &= 1;
+
                             m_MfxVideoParams.IOPattern |= pParam->bStoreMetaData ? MFX_IOPATTERN_IN_VIDEO_MEMORY : MFX_IOPATTERN_IN_SYSTEM_MEMORY;
 
                             mfxU32 mode = pParam->bStoreMetaData ? MfxOmxInputSurfacesPool::MODE_LOAD_MDBUF
-- 
2.17.1

